python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/base_scenario.py

```diff
@@ -68,6 +70,64 @@
 from dagster._daemon.asset_daemon import AssetDaemon
 
 
+def _get_implicit_auto_materialize_policy(
+    asset_graph, asset_key: AssetKey
+) -> Optional[AutoMaterializePolicy]:
+    """For backcompat with pre-auto materialize policy graphs, assume a default scope of 1 day."""
+    auto_materialize_policy = asset_graph.get_auto_materialize_policy(asset_key)
+    if auto_materialize_policy is None:
+        time_partitions_def = get_time_partitions_def(asset_graph.get_partitions_def(asset_key))
+        if time_partitions_def is None:
+            max_materializations_per_minute = None
+        elif time_partitions_def.schedule_type == ScheduleType.HOURLY:
+            max_materializations_per_minute = 24
+        else:
+            max_materializations_per_minute = 1
+        return AutoMaterializePolicy(
+            on_missing=True,
+            on_new_parent_data=not bool(
+                asset_graph.get_downstream_freshness_policies(asset_key=asset_key)
+            ),
+            for_freshness=True,
+            max_materializations_per_minute=max_materializations_per_minute,
+        )
+    return auto_materialize_policy
+
+
+def get_repo_with_implicit_auto_mterialize_policies(
+    assets: Optional[Sequence[Union[SourceAsset, AssetsDefinition]]]
+) -> Sequence[AssetsDefinition]:
+    """Inject auto materialize policies into the assets that don't have one."""
+
+    # hack in order to use asset_graph.get_downstream_freshness_policies
```

Comment:
this feels bad but doesn't really change the runtime for the suite on my machine